### PR TITLE
Server and Client can now perform first 2 steps of handshake

### PR DESCRIPTION
**Pull request by MostlyArmless**
Currently the system is capable of the following steps:
1. server starts up
2. when client starts, it automatically connects and sends a clientIntro message
3. server receives this and automatically replies with a serverRegistration message (though it doesn't yet assign a clientId, it just echoes the clientId sent by the client)

also, the client can still send chat messages to the server but nothing new is happening with them.

There's a ChatRoom class that's been added but it isn't finished yet. It will be what re-broadcasts the messages to the members.

### CODE REVIEW

File: src/backend/ChatAppServer.ts [hunk 1]
```diff
@@ -75,52 +108,18 @@ export class ChatAppServer {
...
-            return;
-        }
-        // console.log(`${msg.time}\t${msg.clientId} to ${msg.chatroom}: ${msg.content}`);
-    }
-
     // TODO handle the case where the client impersonates someone else by making up the clientId (on a clientChatMessage)
     handleChatMessage(msg: IFromClientChatMessage, ws: wsWebSocket) {
         // For now we're just gunna echo the message back to the client, but as a IFromServerChatMessage
-        this.chatrooms
+        const thisChatroom = this.chatrooms.get(msg.chatroom);
```
Comment (JessZanewich) [inline src/backend/ChatAppServer.ts]: I feel like using `this` in your variable name will just cause confusion. this means different things in different contexts, so it is not benefitting your code to have it in the name.

Comment (MostlyArmless) [reply src/backend/ChatAppServer.ts]: Agreed, will fix

Comment (MostlyArmless) [reply src/backend/ChatAppServer.ts]: Do you know what best practice on GitHub is for addressing requested changes during a pull request? Should I just make another commit on this same branch that fixes this and then mark this conversation as closed, or is there a better way?

File: src/chat-app-frontend/src/Components/WebSocketChatBox.tsx [hunk 2]
```diff
@@ -23,13 +24,15 @@ export class WebSocketChatBox extends Component<WebSocketWrapperProps, WebSocket
 	constructor(props: WebSocketWrapperProps) {
 		super(props);
 		this.state = {
+			clientId: 0,
 			websocket: new WebSocket(props.websocketServerUrl),
 			clientMessage: "",
-			receivedMessageHistory: "",
+			receivedMessageHistory: [],
```
Comment (JessZanewich) [inline src/chat-app-frontend/src/Components/WebSocketChatBox.tsx]: This change will unblock my messages ticket! Thank you for the update to an array.

**Pull request merged**